### PR TITLE
Test provider upgrades for popular resources

### PR DESCRIPTION
**Pull request by t0yv0**
Fixes https://github.com/pulumi/home/issues/3132

```
    provider_test.go:44: Resources covered: 34
    provider_test.go:44: - azure:appinsights/insights:Insights
    provider_test.go:44: - azure:appservice/appService:AppService
    provider_test.go:44: - azure:appservice/functionApp:FunctionApp
    provider_test.go:44: - azure:appservice/plan:Plan
    provider_test.go:44: - azure:core/resourceGroup:ResourceGroup
    provider_test.go:44: - azure:dns/aRecord:ARecord
    provider_test.go:44: - azure:dns/txtRecord:TxtRecord
    provider_test.go:44: - azure:dns/zone:Zone
    provider_test.go:44: - azure:keyvault/accessPolicy:AccessPolicy
    provider_test.go:44: - azure:keyvault/keyVault:KeyVault
    provider_test.go:44: - azure:keyvault/secret:Secret
    provider_test.go:44: - azure:management/lock:Lock
    provider_test.go:44: - azure:monitoring/actionGroup:ActionGroup
    provider_test.go:44: - azure:monitoring/metricAlert:MetricAlert
    provider_test.go:44: - azure:network/networkInterface:NetworkInterface
    provider_test.go:44: - azure:network/networkSecurityGroup:NetworkSecurityGroup
    provider_test.go:44: - azure:network/networkSecurityRule:NetworkSecurityRule
    provider_test.go:44: - azure:network/publicIp:PublicIp
    provider_test.go:44: - azure:network/route:Route
    provider_test.go:44: - azure:network/routeTable:RouteTable
    provider_test.go:44: - azure:network/subnet:Subnet
    provider_test.go:44: - azure:network/virtualNetwork:VirtualNetwork
    provider_test.go:44: - azure:servicebus/namespace:Namespace
    provider_test.go:44: - azure:servicebus/queue:Queue
    provider_test.go:44: - azure:servicebus/subscription:Subscription
    provider_test.go:44: - azure:servicebus/subscriptionRule:SubscriptionRule
    provider_test.go:44: - azure:servicebus/topic:Topic
    provider_test.go:44: - azure:sql/database:Database
    provider_test.go:44: - azure:sql/firewallRule:FirewallRule
    provider_test.go:44: - azure:sql/sqlServer:SqlServer
    provider_test.go:44: - azure:storage/account:Account
    provider_test.go:44: - azure:storage/blob:Blob
    provider_test.go:44: - azure:storage/container:Container
    provider_test.go:44: - random:index/randomString:RandomString
```

### CODE REVIEW

**Comment by t0yv0** [comment]
Remarkably getting Test_keyvault fail in CI but not locally. What could be different..

**Comment by t0yv0** [comment]
This test passed once I recorded the snapshot with the same credentials CI uses for testing.

File: .ci-mgmt.yaml [hunk 1]
```diff
@@ -30,3 +30,13 @@ runner:
   prerequisites: pulumi-ubuntu-8core
   buildSdk: pulumi-ubuntu-8core
   publish: pulumi-ubuntu-8core
+actions:
+  preTest:
+
+    - name: make upstream
+      run: |
+        make upstream
```
Comment (iwahbe) [inline .ci-mgmt.yaml:38]: Optional nit: This is equivalent.
```suggestion
    - run: make upstream
```

**Review by iwahbe — APPROVED** [review]
LGTM

**Pull request merged**